adatest/_model.py

```diff
@@ -79,3 +84,56 @@ def __call__(self, strings):
         for s, data in zip(strings, inner_out):
             out.append(data[0]["generated_text"][len(s):]) # remove the input text from the output
         return out
+
+class EntailmentAPI(Model):
+    def __init__(self,model, url, api_key):
+        self._inner_model = APICaller(url, api_key)
+        self.output_names = None
+    
+    def __call__(self,sequences,candidate_labels,hypothesis_template,multi_class= False, th = 0.95):
+        out = []
+        if isinstance(sequences,str):
+            sequences = [sequences]
+        for seq in sequences:
+            if not seq:
+                #manual patch for empty seq generated by neox
+                seq = "[EMPTYSEQ]"
+            inner_out = self._inner_model(seq,candidate_labels,hypothesis_template,multi_class,th)
+            """
+            try:
+                inner_out = self._inner_model(seq,candidate_labels,hypothesis_template)
+            
+            except:
+                inner_out = [[x,np.nan] for x in candidate_labels]
+            """
+            scores = [x[1] for x in inner_out]
+            label_in_score_order = [x[0] for x in inner_out]
+            out.append({'sequence':seq,'labels':label_in_score_order,'scores':scores})
+        return out
+            
+class APICaller():
+
+    def __init__(self,url, api_key):
+        self.url = url
+        self.headers = headers = {'Content-Type':'application/json', 'Authorization':('Bearer '+ api_key), 'azureml-model-deployment': 'entailment-phrases' }
```

Comment:
I'd remove the 'azureml-model-deployment' header so that any requests follow the traffic rules set in Azure ML rather than to a specific deployment (which may not exist for all models).